src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -194,7 +205,7 @@ protected override async Task HandleChallengeAsync(AuthenticationProperties prop
         {
             if (string.IsNullOrEmpty(properties.RedirectUri))
             {
-                properties.RedirectUri = CurrentUri;
+                properties.RedirectUri = OriginalPathBase + Request.Path + Request.QueryString;
```

Comment:
@Tratcher stricto sensu, these changes are not required, but if we end up flowing the `ReturnUrl` up to the access denied endpoint, we'll likely want it to be captured as a relative URL so helpers like `IUrlHelper.IsLocalUrl` and `RedirectToLocal()` work correctly (they don't deal with absolute URLs, even if the domain matches the current one).

Reply:
This should be OriginalPathBase + OriginalPath.

Reply:
I copied it from the cookie handler: https://github.com/aspnet/Security/blob/master/src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs#L425

Should I also update the cookie handler?

Reply:
Sure. FYI: https://github.com/aspnet/Security/issues/1730